### ISSUE TITLE
Logout response to idP initiated SLO

### ISSUE THREAD

**Issue by PaulaSazonov** [node-saml/passport-saml#445]
How to return a logout response to a logout request from idP? I have an idP that requires this as part of the SLO process, and after multiple attempts and going through the source code and documentation, I have not found out how to do this. Any advice would be much appreciated!

**Comment by srd90** [comment]
passport-saml's incoming LogoutRequest handling implementation is not spec compliant (it is unable to terminate SP side session using only information provided in LogoutRequest).
Furthermore it may leave SP side sessions open (if IdP and SP are at different domains and if/when webapplication's session cookie is not delivered due various reasons to webapplication) EVEN THOUGH  it shall always report to IdP that session at SP side was successfully terminated which means that end user may see false information at IdP's logout tracking page.

See issue #419 for further information.

tl;dr; you should not use passport-saml's current SLO handling implementation at all if your IdP is at different domain than your SP. LogoutRequest from IdP to SP over POST binding is especially dangerous/problematic.

**Comment by PaulaSazonov** [comment]
Thank you @srd90 for the insight, much appreciated! Still have not managed to produce a working solution, but it is good to know if the library's implementation is not spec compliant. Also have tried suggestions provided in issue #221, but to no success. Would be great if the documentation offered some more information on the logout process!

**Comment by PaulaSazonov** [comment]
I was able to solve my problem returning a LogoutResponse, and thought I'd share in case someone else is struggling with the same thing: I had configured my strategy `authnRequestBinding: 'HTTP-POST'` and `skipRequestCompression: true`, and this caused issues with the redirect endpoints (including a problem with `strategy.logout()`). Using a strategy without these flags set solved these issues, and I was able to return a valid LogoutResponse to the IdP. I would say this issue relates to #241, which was why the `skipRequestCompression: true` flag was set in the first place.

**Comment by markstos** [comment]
Closing as dupe of #419.
